### ISSUE TITLE
Torrent-stream does not detect files 

### ISSUE THREAD

**Issue by tsoernes** [mafintosh/torrent-stream#142]
Hi.

I have a simple function:
```
function play(magnet) {
	var torrentStream = require('torrent-stream');
	console.log("Creating torrent stream for " + magnet);
	var engine = torrentStream(magnet);
	
	engine.on('ready', function() {
		engine.files.forEach(function(file) {
			console.log('filename:', file.name);
			var stream = file.createReadStream();
			// stream is readable stream to containing the file content
		});
	});
}
```
to which I pass this link: `magnet:?xt=urn:btih:02b6b3ed991edb7f173469f5bf8d8267
7380ea53&tr=http://tracker.torrentleech.org:2710/a/daaec160fe1144f9a01ec77260160
dfc/announce`.

The link works If i open it manually in a torrent program, but `console.log('filename:', file.name);` does not print anything. Why?

**Comment by mafintosh** [comment]
Try console.log'ing in the ready handler and see if the event is fired

**Comment by tsoernes** [comment]
Like this `engine.on(console.log('ready'), function()` ? Yes, it is fired.

**Comment by tsoernes** [comment]
console.log('filename:', file.name) should fire even if the all the files are in a top-level directory, right? If not, how to get the files inside the directory?

**Comment by tsoernes** [comment]
@mafintosh what does it mean that the ready handler event is fired, but that no files show up? It's a private tracker, if that means anything.

**Comment by mafintosh** [comment]
@tsoernes does this fire?

``` js
engine.on('ready', function () {
  console.log('ready')
})
```

**Comment by tsoernes** [comment]
No. I also tried replacing : and / in the magnet link with %3A and %2F, but no luck.

**Comment by tsoernes** [comment]
@mafintosh the tracker FAQ says: "You have DHT, Peer Exchange, Local Peer Discovery or DNA turned on in your client. All those settings have to be turned OFF. " I tried turning DHT off, but no luck: the ready event is not fired, but execution finishes (it does not with dht:true).

I will also add that dragging and dropping torrent files from this particular tracker *does work* with popcorn-time, which uses peer-flix.

**Comment by tsoernes** [comment]
Take a look:
DHT Off:
![Uploading tlscrape1.png…]()
DHT On (never finishes):
![tlscrape2](https://cloud.githubusercontent.com/assets/6782404/10869603/94f6f51c-80b4-11e5-95c1-71cc033a1843.png)

**Comment by williamsbs** [comment]
hi i still have this probleme, some magnets don't trigger the engine.on('ready', fn),
does anyone have a solution ?
